modules/swagger-codegen/src/main/java/io/swagger/codegen/languages/RustClientCodegen.java

```diff
@@ -0,0 +1,515 @@
+    static Logger LOGGER = LoggerFactory.getLogger(RustClientCodegen.class);
+
+    protected String packageName = "swagger";
+    protected String packageVersion = "1.0.0";
+    protected String apiDocPath = "docs/";
+    protected String modelDocPath = "docs/";
+    protected String apiFolder = "src/apis";
+    protected String modelFolder= "src/models";
+
+    public CodegenType getTag() {
+        return CodegenType.CLIENT;
+    }
+
+    public String getName() {
+        return "rust";
+    }
+
+    public String getHelp() {
+        return "Generates a Rust client library (beta).";
+    }
+
+    public RustClientCodegen() {
+        super();
+        outputFolder = "generated-code/rust";
+        modelTemplateFiles.put("model.mustache", ".rs");
+        apiTemplateFiles.put("api.mustache", ".rs");
+
+        modelDocTemplateFiles.put("model_doc.mustache", ".md");
+        apiDocTemplateFiles.put("api_doc.mustache", ".md");
+
+        embeddedTemplateDir = templateDir = "rust";
+
+        setReservedWordsLowerCase(
+            Arrays.asList(
+				"abstract", "alignof", "as", "become", "box",
+				"break", "const", "continue", "crate", "do",
+				"else", "enum", "extern", "false", "final",
+				"fn", "for", "if", "impl", "in",
+				"let", "loop", "macro", "match", "mod",
+				"move", "mut", "offsetof", "override", "priv",
+				"proc", "pub", "pure", "ref", "return",
+				"Self", "self", "sizeof", "static", "struct",
+				"super", "trait", "true", "type", "typeof",
+				"unsafe", "unsized", "use", "virtual", "where",
+				"while", "yield"
+            )
+        );
+
+        defaultIncludes = new HashSet<String>(
+                Arrays.asList(
+                    "map",
+                    "array")
+                );
+
+        languageSpecificPrimitives = new HashSet<String>(
+            Arrays.asList(
+				"i8", "i16", "i32", "i64",
+				"u8", "u16", "u32", "u64",
+				"f32", "f64",
+				"char", "bool")
+            );
+
+        instantiationTypes.clear();
+        /*instantiationTypes.put("array", "GoArray");
+        instantiationTypes.put("map", "GoMap");*/
+
+        typeMapping.clear();
+        typeMapping.put("integer", "i32");
+        typeMapping.put("long", "i64");
+        typeMapping.put("number", "f32");
+        typeMapping.put("float", "f32");
+        typeMapping.put("double", "f64");
+        typeMapping.put("boolean", "bool");
+        typeMapping.put("string", "String");
+        typeMapping.put("UUID", "String");
+        typeMapping.put("date", "string");
+        typeMapping.put("DateTime", "String");
+        typeMapping.put("password", "String");
+        typeMapping.put("file", "File");
+        // map binary to string as a workaround
+        // the correct solution is to use []byte
+        typeMapping.put("binary", "String");
+        typeMapping.put("ByteArray", "String");
+        typeMapping.put("object", "Object");
+
+        importMapping = new HashMap<String, String>();
```

Comment:
Not needed in rust, we can import everything by absolute paths, also saves the reserved namespace from pollution.